### PR TITLE
workspace loader for better UX

### PR DESCRIPTION
**Pull request by sulaimon23**
# Fixes Issue/Linear Ticket

( This could be an existing issue or a linear ticket )

**My PR Closes #issue_number_here** or
**My PR Fixes ID-linear_ticket_number_here**

# Changes proposed

## What did you do?

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x

https://user-images.githubusercontent.com/61692103/206876180-39b5c1dc-78ae-41b1-83de-88580a96848b.mov

] I am only making changes to files I was requested to.

# Screenshots/ Videos

### CODE REVIEW

File: packages/main/src/components/general-loading/WorkspaceLoading.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import ZuriChatLogo from "../../assets/zuri-chat-logo/logo.png";
+
+const WorkspaceLoading = ({ text }) => {
+  //Redirect to dashboard
+  return (
+    <div className={styles.loadingContainer}>
+      <div className={styles.loaderContainer}>
+        <div>
+          <img
+            width="150"
```
Comment (billmal071) [inline packages/main/src/components/general-loading/WorkspaceLoading.jsx:13]: please add alt to this.

File: packages/main/src/components/general-loading/WorkspaceLoading.module.css [hunk 2]
```diff
@@ -0,0 +1,61 @@
+* {
+    padding: 0;
+    margin: 0;
+    box-sizing: border-box;
+}
+
+.loadingContainer {
+    /* position: fixed; */
```
Comment (billmal071) [inline packages/main/src/components/general-loading/WorkspaceLoading.module.css:8]: Remove commented out code.

File: packages/main/src/components/general-loading/WorkspaceLoading.module.css [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    background: #fff;
+    top: 0;
+    left: 0;
+    height: 100vh;
+    width: 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-direction: column;
+    /* z-index: 999999999; */
```
Comment (billmal071) [inline packages/main/src/components/general-loading/WorkspaceLoading.module.css:18]: remove commented out code.

File: packages/main/src/pages/protected/workspace/Workspace.jsx [hunk 4]
```diff
@@ -22,6 +22,7 @@ import VideoChat from "../../../components/media-chat/VideoChat";
 import VoiceCall from "../../../components/media-chat/VoiceCall/VoiceCall";
 import { Sidebar, TopBar } from "../../../components/protected";
 import ExtraSidebar from "../../../components/protected/extra-sidebar";
+import WorkspaceLoading from "../../../components/general-loading/WorkspaceLoading";
 
 export default function Index() {
   // const { workspaceId } = useParams();
```
Comment (billmal071) [inline packages/main/src/pages/protected/workspace/Workspace.jsx:28]: You can remove this if it's not being used.

File: packages/root-config/src/zuri-root-config.js [hunk 5]
```diff
@@ -31,11 +42,10 @@ window.addEventListener("zuri-plugin-load", () => {
...
           name: plugin.name,
-          app: () => System.import(`@zuri/${plugin.name}`),
+          app: loadApp(plugin),
           activeWhen: pathToActiveWhen(`/workspace/:id/${plugin.pluginPath}`),
           customProps: {
             domElement: document.getElementById("zuri-plugin-load-section"),
-
             baseName: `/workspace/${currentWorkspace}/${plugin.pluginPath}`
             // baseName: `/workspace/${localStorage.getItem("currentWorkspace")}/${
             //   plugin.pluginPath
```
Comment (billmal071) [inline packages/root-config/src/zuri-root-config.js:51]: You can remove.

**Review by billmal071 — CHANGES_REQUESTED** [review]

File: packages/main/src/components/general-loading/WorkspaceLoading.module.css (hunk 3 above)
Comment (sulaimon23) [reply packages/main/src/components/general-loading/WorkspaceLoading.module.css]: Done

File: packages/main/src/components/general-loading/WorkspaceLoading.module.css (hunk 2 above)
Comment (sulaimon23) [reply packages/main/src/components/general-loading/WorkspaceLoading.module.css]: Done

File: packages/main/src/components/general-loading/WorkspaceLoading.jsx (hunk 1 above)
Comment (sulaimon23) [reply packages/main/src/components/general-loading/WorkspaceLoading.jsx:13]: Done

File: packages/main/src/pages/protected/workspace/Workspace.jsx (hunk 4 above)
Comment (sulaimon23) [reply packages/main/src/pages/protected/workspace/Workspace.jsx]: Done

File: packages/root-config/src/zuri-root-config.js (hunk 5 above)
Comment (sulaimon23) [reply packages/root-config/src/zuri-root-config.js]: Done

**Pull request merged**